lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/remote/transport/netty/NettyMessagingTransport.java

```diff
@@ -176,6 +179,26 @@ public NettyMessagingTransport(final String hostAddress, int port,
     LOG.log(Level.FINE, "Starting netty transport socket address: {0}", this.localAddress);
   }
 
+    /**
+     * Constructs a messaging transport
+     *
+     * @param hostAddress   the server host address
+     * @param port          the server listening port; when it is 0, randomly assign a port number
+     * @param clientStage   the client-side stage that handles transport events
+     * @param serverStage   the server-side stage that handles transport events
+     * @param numberOfTries the number of tries of connection
+     * @param retryTimeout  the timeout of reconnection
+     */
+    @Deprecated
```

Comment:
The indents seem off. Also, add a `@deprecated` inside of the JavaDoc to inform users what to use instead.